### DISCUSSION TITLE
Font color ignored for button toggled state

### DISCUSSION THREAD

**Discussion by nagyrobi** [HASwitchPlate/openHASP#108] [Q&A]
[text_color](https://fvanroie.github.io/hasp-docs/#styling/#text) for button in toggled (on) state is ignored when set through MQTT.
UI theme is `Hasp Light` set in web interface.

Is this intentional?

**Comment by adorobis** [comment]
I have similar observation for `bg_color` and `bg_grad_color`. In `0` (off) state they take the value correctly but in toggled `1` (on) state they display theme color (Hasp Dark in my case).

**Comment by fvanroie** [comment]
Is it still the case? I tweaked the theme a bit for consistency: cfcaa16ab3856a5f3801c5cd0f8977600ba7142b

**Reply by nagyrobi** [reply]
Yes, unfortunately: https://github.com/HASwitchPlate/openHASP/issues/295

**Comment by nagyrobi** [comment]
Here's the trick (thanks @fvanroie). Add these to the _jsonl_ of the button:
`"text_color":"white","text_color2":"white","text_color1":"yellow","text_color3":"yellow"`
No need to set the colors dinamycally through MQTT!

**Reply by adorobis** [reply]
What if I still need to set the colors dynamically e.g. because I want them based on a different entity state? Will it work?

**Reply by nagyrobi** [reply]
Feel free to discover them yourself by experimenting. Try setting dynamically only `text_color1` and `text_color3`.

**Reply by adorobis** [reply]
OK, will do in free time :)
